### PR TITLE
Add Prettier pre-commit hook

### PR DESCRIPTION
**Pull request by pscosta5**
Additions:

- Prettier pre-commit hook
  - Added `prose-wrap` option and set `print-width` to 88 to match Black styling

- Formatted all Markdown and YAML files to be compliant to the pre-commit rules

### CODE REVIEW

**Comment by cjolowicz** [comment]
Thanks so much for this!

Markdown and reST files use semantic line breaks, so I’m not sure the prosewrap option is the way to go here though.

https://sembr.org/

**Comment by pscosta5** [comment]
@cjolowicz That's fair. I think that would be a bit difficult to enforce in a project—but I get where you're coming from.

I've updated prettier to preserve line breaks as is.

These changes are reflected in the repo's Markdown files!

File: {{cookiecutter.project_name}}/.github/workflows/test-pypi.yml [hunk 1]
```diff
@@ -7,19 +7,18 @@ jobs:
...
-        repository_url: https://test.pypi.org/legacy/
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+          architecture: x64
+      - run: pip install poetry==1.0.3
+      - run: >-
+          poetry version patch && version=$(poetry version | awk '{print $2}') && poetry
+          version $version.dev.$(date +%s)
```
Comment (cjolowicz) [inline {{cookiecutter.project_name}}/.github/workflows/test-pypi.yml]:
```suggestion
          poetry version patch &&
          version=$(poetry version | awk '{print $2}') &&
          poetry version $version.dev.$(date +%s)
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -5,18 +5,15 @@ jobs:
...
-        cookiecutter==1.7.0
-    - run: cookiecutter --no-input .
-    - run: cd hypermodern-python && nox
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+      - run: >-
+          pip install nox==2019.11.9 poetry==1.0.3 cookiecutter==1.7.0
```
Comment (cjolowicz) [inline .github/workflows/tests.yml]:
```suggestion
          pip install
          nox==2019.11.9
          poetry==1.0.3
          cookiecutter==1.7.0
```

**Comment by pscosta5** [comment]
Apologies for missing those! Thanks for fixing my mistakes!

**Pull request merged**